### PR TITLE
[DEV-12462] - handle pending mfa requests

### PR DESCRIPTION
**Pull request by tomerstein**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -46,3 +60,36 @@ func GetOAuthAuthURL(appURL string) string {
 func GetOAuthTokenURL(appURL string) string {
 	return fmt.Sprintf("%s/oauth/token", appURL)
 }
+
+func GetProfileByName(profileName ProfileName) (*SessionConfig, error) {
```
Comment (tomerstein) [inline pkg/config/config.go:64]: moved from `pkg/aponoapi/client_factory.go`

File: pkg/services/requests.go [hunk 2]
```diff
@@ -63,6 +68,32 @@ func PrintAccessRequests(cmd *cobra.Command, requests []clientapi.AccessRequestC
...
+	if err != nil {
+		return err
+	}
+
+	portalURL := currentConfig.PortalURL
+	if portalURL == "" {
+		portalURL = config.PortalDefaultURL
+	}
+
+	link := fmt.Sprintf("%s/requests/open", portalURL)
```
Comment (tomerstein) [inline pkg/services/requests.go:82]: will change after Front decide on the correct path

File: pkg/commands/requests/actions/describe.go [hunk 3]
```diff
@@ -42,6 +42,13 @@ func Describe() *cobra.Command {
 				return err
 			}
 
+			if services.IsRequestWaitingForMFA(resp) && *format == utils.TableFormat {
```
Comment (DmitriyO) [inline pkg/commands/requests/actions/describe.go:45]: what if not table format?

File: pkg/commands/apono/runner.go [hunk 4]
```diff
@@ -129,6 +130,7 @@ func createRootCommand(versionInfo version.VersionInfo) *cobra.Command {
 		cmd.SetContext(version.CreateVersionContext(cmd.Context(), &versionInfo))
 		cmd.SetContext(analytics.CreateStartTimeContext(cmd.Context(), &commandStartTime))
 		cmd.SetContext(analytics.CreateCommandIDContext(cmd.Context(), commandID))
+		cmd.SetContext(config.CreateProfileContext(cmd.Context(), profileName))
```
Comment (DmitriyO) [inline pkg/commands/apono/runner.go:133]: what is this for?

File: pkg/services/requests.go [hunk 5]
```diff
@@ -63,6 +68,39 @@ func PrintAccessRequests(cmd *cobra.Command, requests []clientapi.AccessRequestC
...
+	var prefixMessage string
+	if requestID != nil {
+		prefixMessage = fmt.Sprintf("Request %s", color.Bold.Sprint(*requestID))
+	} else {
+		prefixMessage = "Some requests"
+	}
+
+	_, err = fmt.Fprintf(
+		cmd.OutOrStdout(),
+		"\n%s %s requires completing MFA to proceed. It only takes a minute and helps keep you account secure: %s\n",
```
Comment (DmitriyO) [inline pkg/services/requests.go:92]: X -> Some requests requires
lets fix copy adjustments

maybe "require/s"

File: pkg/config/context.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	"errors"
+)
+
+type contextKey string
+
+const (
+	currentProfileContextKey = contextKey("__current_profile")
+)
+
+func CreateProfileContext(ctx context.Context, profileName string) context.Context {
```
Comment (DmitriyO) [inline pkg/config/context.go:14]: what is that?

**Review by DmitriyO — CHANGES_REQUESTED** [review]
Cool, some Q

File: pkg/commands/apono/runner.go (hunk 4 above)
Comment (tomerstein) [reply pkg/commands/apono/runner.go:133]: i need to command current profile in the state for getting the Portal URL

File: pkg/config/context.go (hunk 6 above)
Comment (tomerstein) [reply pkg/config/context.go:14]: i need to command current profile in the state for getting the Portal URL

This function save the current profile to the context

**Review by DmitriyO — APPROVED** [review]

**Pull request merged**